src/lcm/dispatchers.py

```diff
@@ -40,42 +50,53 @@ def spacemap(func, dense_vars, sparse_vars, *, dense_first):
             above but there might be additional dimensions.
 
     """
-    func = allow_args(func)  # vmap cannot deal with keyword-only arguments
-
+    # Check inputs
+    # ==================================================================================
     if not set(dense_vars).isdisjoint(sparse_vars):
```

Comment:
```suggestion
    if overlap := set(dense_vars).intersection(set(sparse_vars)):
```
*(untested)* Then print the problematic stuff in the error message. Same for the other two checks. I know it's pedantic, but this will help tremendously for newcomers.